src/treq/client.py

```diff
@@ -301,17 +306,25 @@ def _request_body(self, data, files, json, stacklevel):
             else:
                 data = []
 
+            content_type = _get_content_type_from_header(
+                headers,
+                b'multipart/form-data; boundary=' + boundary
+            )
             return (
                 multipart.MultiPartProducer(data + files, boundary=boundary),
-                b'multipart/form-data; boundary=' + boundary,
+                content_type,
```

Comment:
Won't setting the content type in this case make the request body unparseable for lack of the `boundary` parameter?